### ISSUE TITLE
RHYX-M21-45 camera boot issues

### ISSUE THREAD

**Issue by tomu4** [prusa3d/Prusa-Firmware-ESP32-Cam#75]
I got this ESP32-S3-CAM from Aliexpress as a bundle with RHYX-M21-45 camera (according to the description, it should have been OV2640, but that's not the point). I tried loading compiled firmware, which caused bootloop. I tried compiling my own from source files, and got camera initialization error 106, as I found online, it's 0x106 - Operation or feature not supported. I therefore changed   CameraConfig.pixel_format in camera.cpp from   PIXFORMAT_JPEG to PIXFORMAT_RGB565. That solved the bootloop issue, I am able to load the configuration website with no problem, but it doesn't show an image, neither is the image uploaded to Prusa Connect. I suppose it's because camera's lack of JPEG support. Is there more troubleshooting I could try, or do I just have to buy original OV2640 camera with JPEG support?

**Comment by alfs** [comment]
The RHYX labeled camera seems to be a GC2145, not OV2640.
Also you may need to limit resolution to QVGA or QQVGA.

**Comment by tomu4** [comment]
> The RHYX labeled camera seems to be a GC2145, not OV2640.
Yeah, I figured it's not OV2640 (despite labeled as so). I just wanted to know whether this particular camera was even supported. 
> Also you may need to limit resolution to QVGA or QQVGA.
You mean in addition to changing format, or keep format set to JPEG and limit resolution?

**Comment by alfs** [comment]
> > The RHYX labeled camera seems to be a GC2145, not OV2640.
> 
> Yeah, I figured it's not OV2640 (despite labeled as so). I just wanted to know whether this particular camera was even supported.
> 
> > Also you may need to limit resolution to QVGA or QQVGA.
> 
> You mean in addition to changing format, or keep format set to JPEG and limit resolution?

It doesn't support compression (so no JPEG at all), meaning that higher resolutions may require too long transmission time. You can of course try higher resolutions - just something to try/consider if it isn't working.

**Comment by tomu4** [comment]
Well, I solved it by buying OV2640 instead. Even if I got it to work, the quality would suck anyway. I just think there should be updated documentation for this particular camera type, since it's exactly this one that's on photos in github documentation, and if it doesn't work out of the box, there should at least be information on how to make it work. Anyway, for me, the problem is solved, so I'm going to close the issue, but I believe someone else is for sure going to have the same problem.

**Comment by spyder12342** [comment]
![Screenshot 2024-12-19 174347](https://github.com/user-attachments/assets/4bc318af-1dbe-484c-a159-e4203e9f8ea4)
I am using qv260 but i am having the same issue

**Comment by Robotiox** [comment]
I also got that problem. Please can someone help me?

**Comment by gtelzur** [comment]
Me too... I decided to order the one that is known to work, i.e. OV2640,  and not a poor substitute

**Comment by jayraj2827** [comment]
But what do if someone has that camera module someone have Any Solution for this??

**Comment by manojcode4** [comment]
Bro I also face some error but I try in rgb565 image save in sd card but not support to display

**Comment by sabinbaba** [comment]
but if change that camera, and i use OV2640 instead of using  GC2145  will it work ??

**Comment by tomu4** [comment]
> but if change that camera, and i use OV2640 instead of using GC2145 will it work ??

Yes, OV2640 works as intended 🙂
